src/Umbraco.Web/Editors/ContentControllerBase.cs

```diff
@@ -102,18 +109,19 @@ protected virtual void MapPropertyValues<TPersisted>(ContentBaseItemSave<TPersis
                 };
 
                 // let the editor convert the value that was received, deal with files, etc
-                var value = valueEditor.FromEditor(data, property.GetValue());
+                var value = valueEditor.FromEditor(data, getPropertyValue(contentItem, property));
 
                 // set the value - tags are special
                 var tagAttribute = propertyDto.PropertyEditor.GetTagAttribute();
                 if (tagAttribute != null)
                 {
                     var tagConfiguration = ConfigurationEditor.ConfigurationAs<TagConfiguration>(propertyDto.DataType.Configuration);
                     if (tagConfiguration.Delimiter == default) tagConfiguration.Delimiter = tagAttribute.Delimiter;
+                    //fixme how is this supposed to work with variants?

```

Comment:
This isn't so much a UI/UX thing, its more about storing data. Tag support will require quite a large undertaking because tags when published are stored in their own DB table for lookup on the front-end. This means that we'll have to store tags with their variant/segment Ids too. So currently it will just not work at all.